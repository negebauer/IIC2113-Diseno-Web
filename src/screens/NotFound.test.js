import React from 'react'
import createHistory from 'history/createBrowserHistory'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import NotFound from './NotFound'
import configureStore from '../redux/store'

let history
let initialState
let store

beforeEach(() => {
  history = createHistory()
  initialState = {}
  const api = { goBack: () => {} }
  store = configureStore(initialState, { history, api })
})

it('shallow renders without crashing', () => {
  shallow(<NotFound store={store} />)
})

it('mount renders without crashing', () => {
  mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <NotFound />
      </ConnectedRouter>
    </Provider>
  )
})

it('matches snapshot', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <NotFound />
      </ConnectedRouter>
    </Provider>
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('doesnt render children', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <NotFound />
      </ConnectedRouter>
    </Provider>
  )
  expect(wrapper.children().exists()).toBeTruthy()
})
