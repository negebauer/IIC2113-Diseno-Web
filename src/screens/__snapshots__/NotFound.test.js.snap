// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectedRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(NotFound)>
        <NotFound
          goBack={[Function]}
        >
          <Styled(styled.div)>
            <styled.div
              className="sc-bZQynM hNKQSy"
            >
              <div
                className="sc-bZQynM hNKQSy sc-ifAKCX kFClOg"
              >
                <h1>
                  404
                </h1>
                <h3>
                  Uh.. parece que te perdiste
                </h3>
                <Component
                  onClick={[Function]}
                >
                  <styled.div>
                    <div
                      className="sc-bxivhb cVWkUG"
                    >
                      <Link
                        onClick={[Function]}
                      >
                        <styled.div
                          onClick={[Function]}
                        >
                          <div
                            className="sc-bdVaJa etGGGl"
                            onClick={[Function]}
                          >
                            Volver a página anterior
                          </div>
                        </styled.div>
                      </Link>
                    </div>
                  </styled.div>
                </Component>
                <Component
                  to="/"
                >
                  <styled.div>
                    <div
                      className="sc-bxivhb cVWkUG"
                    >
                      <Link
                        to="/"
                      >
                        <Styled(Link)
                          to="/"
                        >
                          <Link
                            className="sc-htpNat ipJcUy"
                            replace={false}
                            to="/"
                          >
                            <a
                              className="sc-htpNat ipJcUy"
                              href="/"
                              onClick={[Function]}
                            >
                              Ir a página de inicio
                            </a>
                          </Link>
                        </Styled(Link)>
                      </Link>
                    </div>
                  </styled.div>
                </Component>
              </div>
            </styled.div>
          </Styled(styled.div)>
        </NotFound>
      </Connect(NotFound)>
    </Router>
  </ConnectedRouter>
</Provider>
`;
